src/Simplex/Messaging/Protocol.hs

```diff
@@ -1044,13 +1056,18 @@ class ProtocolMsgTag (Tag msg) => ProtocolEncoding err msg | msg -> err where
 instance PartyI p => ProtocolEncoding ErrorType (Command p) where
   type Tag (Command p) = CommandTag p
   encodeProtocol v = \case
-    NEW rKey dhKey auth_ -> case auth_ of
-      Just auth
-        | v >= 5 -> new <> e ('A', auth)
-        | otherwise -> new
-      _ -> new
+    NEW rKey dhKey subMode auth_
+      | v >= 6 -> new <> subModeEnc <> authEnc
+      | v == 5 -> new <> authEnc
```

Comment:
```suggestion
      | v >= 6 -> new <> e ('S', subMode) <> e ('A', auth)
      | v == 5 -> new <> e ('A', auth)
```

The point of not refactoring it here so it becomes self-documented and you don't have to go through the code to understand the protocol encoding

Reply:
And you need Encoding instance so it works.